docker/Dockerfile

```diff
@@ -1,15 +1,26 @@
-FROM nvcr.io/nvidia/tritonserver:22.12-py3
+FROM nvidia/cuda:12.4.1-devel-ubuntu22.04
 
-RUN rm /etc/apt/sources.list.d/cuda*.list && apt-get update && apt-get install -y --no-install-recommends \
-    rapidjson-dev libgoogle-glog-dev gdb python3.8-venv \
-    && rm -rf /var/lib/apt/lists/* && cd /opt && python3 -m venv py38
+ARG PYTHON_VERSION=3.10
 
-ENV PATH=/opt/py38/bin:$PATH
+RUN rm /etc/apt/sources.list.d/cuda*.list && apt-get update -y && apt-get install -y software-properties-common wget vim &&\
+    add-apt-repository ppa:deadsnakes/ppa -y && apt-get update -y && apt-get install -y --no-install-recommends \
+    rapidjson-dev libgoogle-glog-dev gdb python${PYTHON_VERSION} python${PYTHON_VERSION}-dev python${PYTHON_VERSION}-venv \
+    && apt-get clean -y && rm -rf /var/lib/apt/lists/* && cd /opt && python3 -m venv py3
+
+ENV PATH=/opt/py3/bin:$PATH
 
 RUN python3 -m pip install --no-cache-dir --upgrade pip setuptools==69.5.1 &&\
-    python3 -m pip install --no-cache-dir torch==2.1.0 torchvision==0.16.0 --index-url https://download.pytorch.org/whl/cu118 &&\
+    python3 -m pip install --no-cache-dir torch==2.3.0 torchvision==0.18.0 --index-url https://download.pytorch.org/whl/cu121 &&\
     python3 -m pip install --no-cache-dir cmake packaging wheel
 
+# install openmpi
+RUN wget https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-4.1.5.tar.gz &&\
```

Comment:
what's the benchmark result if we use this docker image?

Reply:
Can the issue #2166 be reproduced in this docker image?

Reply:
Can the TP work?
Anyway, the full test should be conducted before

Reply:
> Can the issue #2166 be reproduced in this docker image?

No. runs successfully on tp=2
```
To find out how much Janet makes every day at the farmers' market, we need to follow these steps:

1. Calculate the total number of eggs laid by Janet's ducks per day: 16 eggs/day
2. Calculate the number of eggs Janet consumes for breakfast and bakes into muffins: 
   - Breakfast: 3 eggs/day
   - Baking: 4 eggs/day
   Total eggs used: 3 + 4 = 7 eggs/day
3. Calculate the number of eggs remaining after Janet consumes and bakes: 
   - Remaining eggs: 16 (total eggs) - 7 (eggs used) = 9 eggs/day
4. Calculate the total amount Janet makes from selling the remaining eggs at the farmers' market: 
   - Price per egg: $2
   - Number of eggs sold: 9 eggs/day
   - Total amount made: 9 eggs/day * $2/egg = $18/day

Therefore, Janet makes $\boxed{$18}$ every day at the farmers' market.
```

Reply:
> Can the TP work? Anyway, the full test should be conducted before
Yes. Tested with lmdeploy chat xx --tp 2